experimental/language-generation/csharp_dotnetcore/05.a.multi-turn-prompt-with-language-fallback/MultiLingualTemplateEngine.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Builder.LanguageGeneration;
+using ActivityBuilder = Microsoft.Bot.Builder.Dialogs.Adaptive.Generators.ActivityGenerator;
+using System;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class MultiLingualTemplateEngineManager
+    {
+        public Dictionary<string, TemplateEngine> TemplateEnginesPerLocale { get; set; } = new Dictionary<string, TemplateEngine>();
+        private LanguagePolicy LangFallBackPolicy;
+
+        public MultiLingualTemplateEngineManager(Dictionary<string, List<string>> lgFilesPerLocale)
+        {
+            if (lgFilesPerLocale == null)
+            {
+                throw new ArgumentNullException(nameof(lgFilesPerLocale));
+            }
+
+            InternalConstructor(lgFilesPerLocale);
+        }
+
+        public Activity GenerateActivityForLocale(string templateName, object data, WaterfallStepContext stepContext)
+        {
+            if (templateName == null)
+            {
+                throw new ArgumentNullException(nameof(templateName));
+            }
+
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            if (stepContext == null || stepContext.Context == null || stepContext.Context.Activity == null)
+            {
+                throw new ArgumentNullException(nameof(stepContext));
+            }
+            return InternalGenerateActivityForLocale(templateName, data, stepContext.Context.Activity.Locale);
+
+        }
+
+        public Activity GenerateActivityForLocale(string templateName, object data, ITurnContext turnContext)
+        {
+            if (templateName == null)
+            {
+                throw new ArgumentNullException(nameof(templateName));
+            }
+
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            if (turnContext == null || turnContext.Activity == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+            return InternalGenerateActivityForLocale(templateName, data, turnContext.Activity.Locale);
```

Comment:
```suggestion
            return GenerateActivityByLocale(templateName, data, turnContext.Activity.Locale);
```

Reply:
It's a little bit cleaner that, we have two "GenerateActivity" overloads which requires different "context" and also provide this “ByLocal" just requires "locale".

Reply:
Fixed.